mlflow/transformers/__init__.py

```diff
@@ -852,6 +866,22 @@ def load_model(
     return _load_model(local_model_path, flavor_config, return_type, device, **kwargs)
 
 
+def _is_model_distributed_in_memory(transformers_model):
+    """Check if the model is distributed across multiple devices in memory."""
+
+    # Check if the model attribute exists and if the model is loaded on CPU
+    if not hasattr(transformers_model, "hf_device_map") or transformers_model.device.type == "cpu":
+        return False
+
+    # Check if all components are loaded into the same location
+    unique_device_types = {value.type for value in transformers_model.hf_device_map.values()}
```

Comment:
Does `value` always have `.type` attribute? It looks like it's possible that `hf_device_map = {'lm_head': 'cpu'}`

Reply:
After doing some more digging into the version changes that they've done to accelerate and transformers, I'm moving away from touching the `hf_device_map` since it's so inconsistent. I moved to a simpler approach for validation of distributed weights files.